wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/api/auth/oauth/OAuthAuthorizationHeaderProvider.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.auth.oauth;
+
+import javax.validation.constraints.NotNull;
+import java.util.Map;
+
+/**
+ * Provides the Authorization header value from the OAuth 1 providers.
+ *
+ * @author Kevin Pollet
+ * @author Igor Vinokur
+ */
+public interface OAuthAuthorizationHeaderProvider {
+
+    /**
+     * Returns the Authorization header value used to sign the request with OAuth.
+     *
+     * @param oauthProviderName
+     *         the OAuth 1 provider name.
+     * @param userId
+     *         the user id.
+     * @param requestMethod
+     *         the HTTP request method.
+     * @param requestUrl
+     *         the HTTP request url with encoded query parameters.
+     * @param requestParameters
+     *         the HTTP request parameters. HTTP request parameters must include raw values of application/x-www-form-urlencoded POST
+     *         parameters.
+     * @return the Authorization header value or {@code null} if it cannot be computed.
+     */
+    String getAuthorizationHeader(@NotNull String oauthProviderName,
```

Comment:
I think `@NotNull` annotations for all paremeters are redundant. Because we have policy that parameters are NotNull by default and you should use @Nullable for nullable parameters.